### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by SebastianBerner**

### CODE REVIEW

File: src/components/ProgressBar.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+class ProgressBar extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    static propTypes = {
+        progress: number.isRequired,
+        progressUpdate: func.isRequired,
```
Comment (andersnylund) [inline src/components/ProgressBar.jsx]: progressUpdate can be removed as it is not used anywhere

File: src/components/ProgressBar.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    className={classes.bar}
+                />
+            </div>
+        );
+    }
+}
+
+ProgressBar.propTypes = {
+    classes: PropTypes.object.isRequired,
+};
```
Comment (andersnylund) [inline src/components/ProgressBar.jsx]: propTypes is defined both here and inside the class as an static property. Choose only one of them

Comment (VonLatvala) [reply src/components/ProgressBar.jsx]: should use shapeOf to tell what kind of object we expect

File: src/components/ProgressBar.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+class ProgressBar extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    static propTypes = {
+        progress: number.isRequired,
+        classes: PropTypes.object.isRequired,
```
Comment (VonLatvala) [inline src/components/ProgressBar.jsx]: https://reactjs.org/docs/typechecking-with-proptypes.html
Search for optionalObjectWithShape on that page.

File: src/store/index.js [hunk 4]
```diff
@@ -18,6 +18,7 @@ const combinedReducers = combineReducers({
     intl: intlReducer,
     answers: answerReducer,
     ui: uiReducer,
+    progress: uiReducer,
```
Comment (valzugg) [inline src/store/index.js]: Same reducer is twice here

File: src/components/QuestionPage.jsx [hunk 5]
```diff
@@ -239,16 +242,15 @@ class QuestionPage extends React.Component {
...
                         direction="column"
                         justify="center"
                         alignItems="center"
                         spacing={16} >
                         <Grid item xs={12} md={12} xl={12}>
                             {this.renderSubmitButton(this.props.questionType)}
-                        </Grid>
-                        <Grid item xs={12} md={12} xl={12}>
+                            <Grid item xs={12} md={12} xl={12}>
+                            </Grid>
```
Comment (valzugg) [inline src/components/QuestionPage.jsx]: Not sure why this was changed but it's what's probably causing the submit and skip buttons to be closer than before, should probably change back to how it was.

File: src/store/index.js (hunk 4 above)
Comment (SebastianBerner) [reply src/store/index.js]: fixed

File: src/components/QuestionPage.jsx (hunk 5 above)
Comment (SebastianBerner) [reply src/components/QuestionPage.jsx]: Yes I changed it back

**Pull request merged**